algorithms/src/Kokkos_Random.hpp

```diff
@@ -1139,33 +1136,29 @@ class Random_XorShift1024_Pool {
  public:
   using generator_type = Random_XorShift1024<DeviceType>;
 
-  KOKKOS_INLINE_FUNCTION
-  Random_XorShift1024_Pool() { num_states_ = 0; }
+#ifdef KOKKOS_ENABLE_DEPRECATED_CODE_4
+  KOKKOS_FUNCTION
+  Random_XorShift1024_Pool() {
+    KOKKOS_IF_ON_DEVICE(
+        /* nonsensical initialization in the name of backward compatibility */
+        num_states_ = 0;)
+    KOKKOS_IF_ON_HOST(init(0, execution_space().concurrency());)
+  }
 
-  inline Random_XorShift1024_Pool(uint64_t seed) {
+  Random_XorShift1024_Pool(uint64_t seed) {
     num_states_ = 0;
 
     init(seed, execution_space().concurrency());
   }
+#else
+  Random_XorShift1024_Pool(uint64_t seed = 0) {
```

Comment:
should this be marked explicit?

Reply:
Resolving as it is a duplicate from another comment above.
I will make sure `Random_XorShift1024_Pool`'s interface is consistent with `Random_XorShift64_Pool`